lib/threads.hpp

```diff
@@ -26,6 +26,7 @@ class GLVisCommand
    VisualizationSceneScalarData **vs;
    StreamState&         curr_state;
    bool                 *keep_attr;
+   SdlWindow            *thread_wnd;
```

Comment:
this probably doesn't matter but if `GLVisCommand` is default-constructed and `signal` is called `thread_wnd` might not be `nullptr` which would fault, consider `nullptr` initializing `thread_wnd` or hide/delete the default ctor

Reply:
I think the default constructor for `GLVisCommand` is implicitly hidden by our defined constructor.

Reply:
You're right! I should have looked it up, sorry about that

ref: https://en.cppreference.com/w/cpp/language/default_constructor